crates/admin/src/cluster_controller/service/state.rs

```diff
@@ -468,10 +440,12 @@ impl TrimMode {
                         debug!(
                             ?partition_id,
                             ?min_applied_lsn,
-                            "Some alive nodes have not applied the log up to the archived LSN"
+                            "Some alive nodes have not applied the log up to the archived LSN; trimming will be delayed until caught up"
                         );
+                        safe_trim_points.insert(log_id, (Lsn::INVALID, *partition_id));
```

Comment:
Unrelated to this PR: One alternative could be to set `archived_lsn.min(min_applied_lsn)` as the safe trim point to harden a bit against the situation where one follower is always a tad bit behind the last taken snapshot.

Reply:
That's definitely a reasonable fallback! I thought I'd play it extra conservative but let's revisit at some point.